include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -62,57 +62,295 @@
 
 namespace RAJA
 {
+/*!
+ * \def RAJA_CUDA_MAX_NUM_BLOCKS
+ * Maximum number of blocks that RAJA will launch
+ */ 
+#define RAJA_CUDA_MAX_NUM_BLOCKS (1024 * 16)
 
-/// Size of reduction memory block for each reducer object (value based on
-/// rough estimate of "worst case" -- need to think more about this...
-#define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
+/*!
+ * \def RAJA_CUDA_REDUCE_BLOCK_LENGTH
+ * Size of reduction memory block for each reducer object (value based on
+ * rough estimate of "worst case" maximum numvber of blocks)
+ */
+#define RAJA_CUDA_REDUCE_BLOCK_LENGTH RAJA_CUDA_MAX_NUM_BLOCKS
 
-/// Reduction Tallies are computed into a small block to minimize UM migration
+/*!
+ * \def RAJA_CUDA_REDUCE_TALLY_LENGTH
+ * Reduction Tallies are computed into a small block to minimize memory motion
+ * Set to Max Number of Reduction Variables
+ */
 #define RAJA_CUDA_REDUCE_TALLY_LENGTH RAJA_MAX_REDUCE_VARS
 
-///
-/// Typedef defining common data type for RAJA-Cuda reduction data blocks
-/// (use this in all cases to avoid type confusion).
-///
-typedef double CudaReductionBlockDataType;
-
-///
-/// Struct used to hold current reduced value and corresponding index
-/// for "loc" reduction operations.
-///
-struct CudaReductionLocBlockDataType {
-  CudaReductionBlockDataType val;
+/*!
+ * \def RAJA_CUDA_REDUCE_VAR_MAXSIZE
+ * Used in CudaReductionDummyDataType to allocate arrays and accommodate all the Reduction types
+ * includes the size of the index variable for Loc reductions
+ */
+#define RAJA_CUDA_REDUCE_VAR_MAXSIZE 16
+
+/*!
+ * \def MACROSTR(x)
+ * Used in the static_assert macros
+ */
+#define STR(x) #x
+#define MACROSTR(x) STR(x)
+
+/*!
+ * \def RAJA_STRUCT_ALIGNAS
+ * abstracts the alignas keyword
+ */
+#define RAJA_STRUCT_ALIGNAS(dalign) alignas(dalign)
```

Comment:
I think there's already an alignment macro in RAJA.hxx.

On 30 Aug 2016, at 8:57, rhornung67 wrote:

> > + * includes the size of the index variable for Loc reductions
>> + */
>> +#define RAJA_CUDA_REDUCE_VAR_MAXSIZE 16
>> +
>> +/*!
>> + * \def MACROSTR(x)
>> + * Used in the static_assert macros
>> + */
>> +#define STR(x) #x
>> +#define MACROSTR(x) STR(x)
>> +
>> +/*!
>> + * \def RAJA_STRUCT_ALIGNAS
>> + * abstracts the alignas keyword
>> + */
>> +#define RAJA_STRUCT_ALIGNAS(dalign) alignas(dalign)
>
> Should this macro also be moved to another header file?
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/LLNL/RAJA/pull/91/files/7bb6c54f379f50473d56e5b7d775d569b061e59e#r76824923